build/source/engine/opSplittin.f90

```diff
@@ -946,8 +946,9 @@ subroutine opSplittin(&
   ! ==========================================================================================================================================
 
   ! success = exit the coupling loop
-  if(ixCoupling==fullyCoupled .and. .not.failure) exit coupling
-
+  if(ixCoupling==fullyCoupled .and. .not. failure) exit coupling   ! terminate DO loop early if fullyCoupled returns a solution
+  if(ixCoupling==stateTypeSplit .and. .not. failure) exit coupling ! terminating the DO loop here is cleaner than letting it complete, because in the latter case the coupling loop will end with ixCoupling = nCoupling+1 = 3 (a FORTRAN loop increments the index variable at the end of each iteration and stops the loop if the index > specified stop value). Variable ixCoupling is used for error reporting in coupled_em.f90 in the balance checks and we thus need to make sure ixCoupling is not incremented to be larger than nCoupling. 
```

Comment:
Could this comment be put above the `if` and then split into a couple of lines to be more readable?

Reply:
Changed as requested.